### PR TITLE
Fixes #99

### LINKED ISSUE #99: Need to add exception handling all over the filesystem for when things are used outside of app
lots of crashing all of the time

### PR DESCRIPTION
**Pull request by shankera**

### CODE REVIEW

File: MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Security.Cryptography;
```
Comment (tevert) [inline MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs]: Why the using Cryptogrophy?

File: MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs [hunk 2]
```diff
@@ -73,6 +74,7 @@ private FileSystem()
                 NotifyFilter = NotifyFilters.LastWrite | NotifyFilters.FileName | NotifyFilters.DirectoryName,
                 IncludeSubdirectories = true
             };
+            _watcher.InternalBufferSize = 32768;
```
Comment (The9thGod) [inline MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs]: Can we abstract 32768 to a constant variable and give a comment explaining why it's 32768 for people other than Tyler and I

File: MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs [hunk 3]
```diff
@@ -567,17 +574,27 @@ private static bool IsFileLocked(FileInfo file)
...
+                _cachedSongs[song.Id].Album = tag.Album;
+                _cachedSongs[song.Id].Genres = tag.Genres;
+                _cachedSongs[song.Id].Year = tag.Year;
+                _cachedSongs[song.Id].TrackNumber = tag.Track;
+                _cachedSongs[song.Id].SupportingArtists = tag.Performers;
+                // Duration is assumed to be fixed
+            }
+            catch (FileNotFoundException)
+            {
+
```
Comment (The9thGod) [inline MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs]: Why aren't we doing anything in this, are these okay?

File: MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs (hunk 2 above)
Comment (shankera) [inline MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs]: yes

File: MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs (hunk 1 above)
Comment (shankera) [inline MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs]: we're not. it shall be cleaned

File: MSOE.MediaComplete/Settings.xaml.cs [hunk 4]
```diff
@@ -100,6 +101,11 @@ private void BtnSelectHomeFolder_Click(object sender, RoutedEventArgs e)
 
             if (!_allDirs.Contains(SettingWrapper.HomeDir.FullPath))
             {
+                var withParent = _allDirs.Where(x => x.StartsWith(SettingWrapper.HomeDir.FullPath, StringComparison.Ordinal));
+                foreach (var parent in withParent)
+                {
+                    _allDirs.Remove(parent);
+                }
```
Comment (tevert) [inline MSOE.MediaComplete/Settings.xaml.cs:109]: I genuinely have no idea what this code is doing. Is it canceling past home dirs that are just subdirectories of the new one? Why?

File: MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs (hunk 3 above)
Comment (shankera) [inline MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs]: as a result of how the changed-renamed-create-delete filewatcher events happen, sometimes we get commands to update a file that has already moved, and therefore it causes an exception. as such, it is unhandled.

Comment (tevert) [inline MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs]: I think a Debug log statement would be a good idea, at a minimum.

File: MSOE.MediaComplete/Settings.xaml.cs (hunk 4 above)
Comment (shankera) [inline MSOE.MediaComplete/Settings.xaml.cs:109]: _alldirs => we have a list of directories that the user has used previously as a root directory. 
if() => if the user selects a directory that already exists in the list, do not do the inner
.where() => returns all the directories the user has previously visited that start with the FULL PATH  of  the newly selected path, then removes them from the list containing all of the previously visited directories

BASICALLY, if you select the parent directory of a directory you've already visited, then we remove the child reference in our stored list

File: MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs [hunk 5]
```diff
@@ -567,17 +577,23 @@ private static bool IsFileLocked(FileInfo file)
...
+                _cachedSongs[song.Id].Title = tag.Title;
+                _cachedSongs[song.Id].Artists = tag.AlbumArtists;
+                _cachedSongs[song.Id].Album = tag.Album;
+                _cachedSongs[song.Id].Genres = tag.Genres;
+                _cachedSongs[song.Id].Year = tag.Year;
+                _cachedSongs[song.Id].TrackNumber = tag.Track;
+                _cachedSongs[song.Id].SupportingArtists = tag.Performers;
+                // Duration is assumed to be fixed
+            }
+            catch (FileNotFoundException) {}
```
Comment (tevert) [inline MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs:595]: Looks like my first comment went away after these lines changed- I think there should be a DEBUG statement in the catch blocks.

Comment (shankera) [inline MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs:595]: This function actually gets called very often. If we had a logger, writing to that would be fine, along with statements of which exception was thrown and when it DOESNT fail, but we would be flooding the console with tons of information, all the time, and it'll all be destroyed when the app stops for that one run. I don't think its worth it at this time

Comment (tevert) [inline MSOE.MediaComplete.Lib/Library/DataSource/FileSystem.cs:595]: Alright, sounds good.
